index.js

```diff
@@ -502,12 +535,20 @@ Router.prototype.use = function use(handler) {
  * and middleware to routes.
  *
  * @param {string} path
+ * @param {string} name (optional)
  * @return {Route}
  * @public
  */
 
-Router.prototype.route = function route(path) {
-  var route = new Route(path)
+Router.prototype.route = function route(path, name) {
+  if(name && this.routes[name]) {
+    throw new Error('a route or handler with that name already exists')
```

Comment:
"with that name" will probably not be too helpful, especially since people are likely to calculate the name. May want to include it in the message?